framework/src/base/Assembly.C

```diff
@@ -555,7 +630,7 @@ Assembly::reinitNeighborAtReference(const Elem * neighbor, const std::vector<Poi
 
     fesd->_phi.shallowCopy(const_cast<std::vector<std::vector<Real> > &>(fe_neighbor->get_phi()));
     fesd->_grad_phi.shallowCopy(const_cast<std::vector<std::vector<RealGradient> > &>(fe_neighbor->get_dphi()));
-    if (_need_second_derivative[fe_type])
+    if (_need_second_derivative.find(fe_type) != _need_second_derivative.end())
```

Comment:
New code commandment:
Thou shall not commit accidental insertion in a std::map by using brackets in a RHS operator unless they can prove it can't fail.